ert3/config/_stages_config.py

```diff
@@ -80,42 +81,36 @@ class TransportableCommand(_StagesConfig):
     _ensure_mime = validator("mime", allow_reuse=True)(_ensure_mime)
 
 
-class Step(_StagesConfig):
+class _Step(_StagesConfig):
     name: str
-    type: Literal["unix", "function"]
-    script: Optional[List[str]] = []
     input: List[InputRecord]
     output: List[OutputRecord]
-    transportable_commands: Optional[List[TransportableCommand]] = []
-    function: Optional[Callable]
-
-    @root_validator
-    def check_defined(cls, step):
-        cmd_names = [cmd.name for cmd in step.get("transportable_commands", [])]
-        script_lines = step.get("script", [])
-        if step.get("type") == "function":
-            if cmd_names:
-                raise ValueError("Commands defined for a function stage")
-            if script_lines:
-                raise ValueError("Scripts defined for a function stage")
-            if not step.get("function"):
-                raise ValueError("No function defined")
-        return step
+
+    # backwards compatible
```

Comment:
What's the rationale for compatibility? Let's just break api now

Reply:
This will need a documentation update on the structure of stages.yml

Reply:
Is there documentation for the infrastructure of stages.yml?
At the moment our documentation just points to stages.yml examples, and that is already been updated.
Is this change to the examples good enough? 
If yes, then we can notify changes in release notes and be done